File: src/main/java/com/flexpoker/game/command/factory/GameFactory.java
package com.flexpoker.game.command.factory;

import java.util.List;

import com.flexpoker.game.command.aggregate.Game;
import com.flexpoker.game.command.commands.CreateGameCommand;
import com.flexpoker.game.command.events.GameEvent;

public interface GameFactory {

    Game createNew(CreateGameCommand command);

    Game createFrom(List<GameEvent> events);

}

File: src/main/kotlin/com/flexpoker/game/command/factory/GameFactory.kt
package com.flexpoker.game.command.factory

import com.flexpoker.game.command.aggregate.Game
import com.flexpoker.game.command.commands.CreateGameCommand
import com.flexpoker.game.command.events.GameEvent

interface GameFactory {
    fun createNew(command: CreateGameCommand): Game
    fun createFrom(events: List<GameEvent>): Game
}